adhocteam/Head-Start-TTADP
# Admin API and schema (2)

**Description of change**
Admin API and database schema for managing users in SmartHub. Adds api routes described in `src/routes/user.js`. Includes database models, relationships and migrations.

This PR also addresses issue #134. Api error handling is centralized and error messages are written to the database.

**How to test**
- run yarn test and verify that tests pass
- alternatively, temporarily replace line 47 in `authMiddleware.js` with `next()` (this will disable auth)
- `yarn deps`
- `yarn db:migrate:undo`
- `yarn db:migrate`
- `yarn server`
- in a client, e.g. Insomnia or Postman set the request `Content-Type` to `application/json` 
and issue requests to the API. e.g. GET `localhost:8080/api/admin/user`
- verify that `200 ` is returned

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/50
* https://github.com/HHS/Head-Start-TTADP/issues/134

**Checklist**

- n/a Meets accessibility standards (WCAG 2.1 Levels A, AA)
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
